### DISCUSSION TITLE
Cost override deprecated and Tracker Tarrif (2 part question)

### DISCUSSION THREAD

**Discussion by Jilas** [BottlecapDave/HomeAssistant-OctopusEnergy#918] [Q&A] [answered]
Firstly I see you are asking about what tariff's people are on, and for the life of me I do not see "tracker" as an option.

And secondly, on upgrading to the new versions v11.1 releases, I get the "Repairs - Cost override deprecated" warning, and I saw the post from another member, that you mentioned finding my "tariff code" in the corresponding current rate sensor.... but I don't have one. 

I can just disable this and problem goes away, but it's useful as its giving the tariff code.  I'm assuming the API is pulling the data from my account as I've never entered it...  Which I can use with some rest pulls direct from Octopus.  Is there a new/other sensor coming that will include my current tariff code Octopus think I am on that either I've missed or is missing for those on tracker.

I am suffering a massive cold/cough, so I may just be missing something really obvious but I've looked at this since the v11.0 release, and I just can't see what I'm missing.

Any help or points would be welcome.  As always Dave thanks for everything you do.

**Comment by Jilas** [comment]
I just had a moment of.. oh is it there (thinking speed is cut to at least a 1/10 normal).  I see the current tariff code is an attribute of the current_rate entity.

Assuming that is not changing, that's fine I can re-code to using that.

So just leaves the question, how different is tracker to the other tariffs and do you want to know how many people are on that (worth noting people can split their services between two tariffs).  And just saw that was posted last year...So maybe a mute point now.

**Reply by BottlecapDave** [reply, marked as answer]
Hello. The tariff code on the cost override is set to your current tariff upon creation. If you ever switched tariff after using the intergation, there's a chance that the restoring of the sensor automatically kept it on your old tariff. This is possibly how you're getting the reparit notice despite never knowingly using the cost override feature. The repair notice is to try and alert people who use it that it's migrating to a new way. If you don't use it, you can ignore the repair notice.

The tariff code on the current rate sensor will always show your current tariff, so will change when you change tariff automatically. This is what you should be using if you need the tariff code for some reason.

>So just leaves the question, how different is tracker to the other tariffs and do you want to know how many people are on that (worth noting people can split their services between two tariffs). And just saw that was posted last year...So maybe a mute point now.

This was a question I asked when I first created the integration so I could get a rough idea on how it's being used and make sure I have testing that covers the various tariffs. OE sometimes have different endpoints or the data comes back in odd ways for some of their tariffs or the tariffs don't fit the norm (e.g. agile or cosy)

**Reply by Jilas** [reply]
Many thanks for replying.   Once my head has shifted whatever is going on with it (if only we could roll back them or have decent virus protection.. Lol), I'll just drop those overrides and then upgrade, now I know where I should be looking for the tariff code. 

Looking back that post was before the tracker became more widely available.   So it makes sense.

Thanks again.
